### PR TITLE
LCH-6543: Add methods for sending client metadata while registering or updating the client.

### PR DESCRIPTION
**Pull request by amangrover90**

### CODE REVIEW

File: test/MetaData/ClientMetaDataTest.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace test\MetaData;
```
Comment (narendradesai) [inline test/MetaData/ClientMetaDataTest.php]:
```suggestion
namespace Acquia\ContentHubClient\test\MetaData;
```

**Review by narendradesai — CHANGES_REQUESTED** [review]
Overall looks good.

File: src/MetaData/ClientMetaData.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Acquia\ContentHubClient\MetaData;
+
+/**
+ * Client metadata.
+ */
+class ClientMetaData {
```
Comment (saxumVermes) [inline src/MetaData/ClientMetaData.php:8]: I like that!

File: src/MetaData/ClientMetaData.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+   * @param string $client_type
+   *   Client type. e.g. drupal, pwa_app, nextjs_app etc.
+   * @param bool $is_publisher
+   *   Whether this site is a publiher or not.
+   * @param bool $is_subscriber
+   *   Whether this site is a subscriber or not.
+   * @param array $client_config
+   *   Additional config metadata.
+   */
+  public function __construct(string $client_type, bool $is_publisher, bool $is_subscriber, array $client_config) {
```
Comment (saxumVermes) [inline src/MetaData/ClientMetaData.php]: Perhaps we can initialise $client_config with an empty array marking it as optional.

Comment (amangrover90) [reply src/MetaData/ClientMetaData.php]: That makes sense.

File: src/MetaData/ClientMetaData.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    throw new \RuntimeException('All the attributes: "client_type", "is_publisher", "is_subscriber" are required.');
+  }
+
+  /**
+   * Returns metadata of the client.
+   *
+   * @return array
+   *   Client metadata.
+   */
+  public function getMetadata(): array {
```
Comment (saxumVermes) [inline src/MetaData/ClientMetaData.php]: Perhaps we can rename this to `toArray`? Or `getValues`? The class is already ClientMetadata, I think it already communicates clearly, that it encapsulates the client metadata.

Comment (amangrover90) [reply src/MetaData/ClientMetaData.php]: Done.

**Review by saxumVermes — APPROVED** [review]
Looks good, I really liked the abstraction of ClientMetadata. I had a tiny concern of it not implementing an interface to comply to the SOLID guidelines, but we can move forward with it I think.

**Review by narendradesai — APPROVED** [review]

**Pull request merged**